Source/Events.Handling/Builder/IEventHandlerMethod.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+
+namespace Dolittle.SDK.Events.Handling.Builder
+{
+    /// <summary>
+    /// Defines an event handler method.
+    /// </summary>
+    public interface IEventHandlerMethod
+    {
+        /// <summary>
+        /// Invokes the event handler method.
+        /// </summary>
+        /// <param name="event">The event.</param>
+        /// <param name="context">The <see cref="EventContext" />.</param>
+        /// <returns>A <see cref="Task" /> representing the asynchronous action.</returns>
+        Task Invoke(object @event, EventContext context);
+    }
+}
```

Comment:
I am wondering why we need both the delegates and this interface (whey they do the same thing). Does it have something to do with what comes next for the EventHandler classes?

Reply:
The delegates represents the two different ways of handling an event (basically either a specific event or an object event) and the IEventHandlerMethod represents an interface that works for both of them. Without having it I don't know how we would achieve the EventHandler class technically because one of the two event handler delegates is generic. If you have suggestions I would like to hear about them :)